osbuild-run

```diff
@@ -5,25 +5,66 @@ import subprocess
 import sys
 
 
-# ld.so.conf must exist, or `ldconfig` throws a warning
-subprocess.run(["touch", "/etc/ld.so.conf"], check=True)
-subprocess.run(["ldconfig"], check=True)
-
-try:
-    subprocess.run(["systemd-sysusers"], stdout=subprocess.PIPE, stderr=subprocess.STDOUT, check=True)
-except subprocess.CalledProcessError as error:
-    sys.stderr.write(error.stdout)
-    sys.exit(1)
-
-# generate /etc/pki/tls/certs/ca-bundle.crt
-os.makedirs("/etc/pki/ca-trust/extracted/pem")
-os.makedirs("/etc/pki/tls/certs")
-os.symlink("/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem", "/etc/pki/tls/certs/ca-bundle.crt")
-subprocess.run(["update-ca-trust"])
-
-# Allow systemd-tmpfiles to return non-0. Some packages want to create
-# directories owned by users that are not set up with systemd-sysusers.
-subprocess.run(["systemd-tmpfiles", "--create"])
-
-r = subprocess.run(sys.argv[1:])
-sys.exit(r.returncode)
+def osbuild_setup_ldconfig():
+    # ld.so.conf must exist, or `ldconfig` throws a warning
+    subprocess.run(["touch", "/etc/ld.so.conf"], check=True)
+    subprocess.run(["ldconfig"], check=True)
+
+
+def osbuild_setup_sysusers():
+    try:
+        subprocess.run(["systemd-sysusers"], stdout=subprocess.PIPE, stderr=subprocess.STDOUT, check=True)
+    except subprocess.CalledProcessError as error:
+        sys.stderr.write(error.stdout)
+        sys.exit(1)
+
+
+def append_certs(cert_conf, dir_fd, parents=b""):
+    for entry in os.scandir(f"/proc/self/fd/{dir_fd}".encode()):
+        if entry.is_file():
+            line = os.path.join(parents, entry.name)
+            print(f"adding: {line}")
+            cert_conf.write(line)
+            cert_conf.write(b"\n")
+        elif entry.is_dir():
+            append_certs(cert_conf, os.open(entry.name, os.O_DIRECTORY, dir_fd=dir_fd), os.path.join(parents, entry.name))
+
+
+def osbuild_setup_certs():
+    # generate /etc/pki/tls/certs/ca-bundle.crt
+    os.makedirs("/etc/pki/ca-trust/extracted/pem")
+    os.makedirs("/etc/pki/tls/certs")
+    os.symlink("/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem", "/etc/pki/tls/certs/ca-bundle.crt")
+
+    try:
+        subprocess.run(["update-ca-trust"])
+    except FileNotFoundError:
+        pass
```

Comment:
I'd like to see a more explicit split between RHEL/Fedora and Debian/Ubuntu. Either *very* explicit by checking `ID_LIKE` in `/etc/os-release`, or by checking which of `update-ca-trust` or `update-ca-certificates`

Reply:
Agreed, but didn't make my mind up about which of those things was best (or if there was something else).

Reply:
Now I made both into separate functions that will just exit silently without doing anything if the binary does not exist. This way we don't need to encode info about different distro's in the tool.